lib/ext/heroku/command/addons.rb

```diff
@@ -447,6 +406,79 @@ def open
 
     private
 
+    def change_plan(direction)
+      addon_name, plan = args.shift, args.shift
+
+      if addon_name && !plan # If invocated as `addons:Xgrade service:plan`
+        deprecate("No add-on name specified (see `heroku help #{current_command}`)")
+
+        addon = nil
+        plan = addon_name
+        service = plan.split(':').first
+
+        action("Finding add-on from service #{service} on app #{app}") do
+          addon = resolve_addon(app, service)
+          addon_name = addon['name']
+        end
+        display "Found #{addon_name} (#{addon['plan']['name']}) on #{app}."
+      else
+        addon_name = addon_name.sub(/^@/, '')
+      end
+
+      raise CommandFailed.new("Missing add-on plan") if plan.nil?
+      raise CommandFailed.new("Missing add-on name") if addon_name.nil?
+
+      labels = {upgrade: 'Upgrading', downgrade: 'Downgrading'}
+      action("#{labels[direction]} #{addon_name} to #{plan}") do
+        api.request(
+          :body     => json_encode({
+            "plan"   => { "name" => plan }
+          }),
+          :expects  => 200..300,
+          :headers  => { "Accept" => "application/vnd.heroku+json; version=edge" },
+          :method   => :patch,
+          :path     => "/apps/#{app}/addons/#{addon_name}"
+        )
+      end
+    end
+
+    def resolve_addon(app_name, service_plan_specifier)
+      service_name, plan_name = service_plan_specifier.split(':')
+
+      # TODO: temporary until heroku/api#3725 is merged
+      service_id = api.request(
+        :expects  => 200,
+        :headers  => { "Accept" => "application/vnd.heroku+json; version=3" },
+        :method   => :get,
+        :path     => "/addon-services/#{service_name}"
+      ).body["id"]
```

Comment:
heroku/api#3725 was merged :boom:

I can't believe I missed this pull-request two weeks ago O.O Thanks, @bjeanes!